src/content/learn/understanding-your-ui-as-a-tree.md

```diff
@@ -0,0 +1,268 @@
+
+These platform UI primitives are not a part of React. React render trees can provide insight to our React app regardless of what platform your app renders to.
+
+</DeepDive>
+
+A render tree represents a single render pass of a React application. With [conditional rendering](/learn/conditional-rendering), a parent component may render different children depending on the data passed.
+
+We can update the Greeting app to conditionally render either an image or a text greeting.
+
+<Sandpack>
+
+```js App.js
+import { Title } from './GreetingMedia';
+import Greeting from './Greeting';
+import Copyright from './Copyright';
+
+export default function App() {
+  return (
+    <>
+      <Title text="Generate a random greeting" />
+      <Greeting>
+        <Copyright year={2004} />
+      </Greeting>
+    </>
+  );
+}
+
+```
+
+```js GreetingMedia.js
+export function Title({fancy, text}) {
+  return <h1 className={fancy ? 'fancy' : ''}>{text}</h1>;
+}
+
+export function Image({src}) {
+  return <img src={src} />;
+}
+```
+
+```js Greeting.js
+import * as React from 'react';
+import greetings from './greetings';
+import {Title, Image} from './GreetingMedia';
+
+function randIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+export default function Greeting({children}) {
+  const [index, setIndex] = React.useState(randIndex(greetings.length));
+  const greeting = greetings[index];
+  const nextGreeting = () => setIndex(randIndex(greetings.length));
+  return (
+    <>
+      {greeting.type === 'text' ? (
+        <Title fancy text={greeting.value} />
+      ) : (
+        <Image src={greeting.value} />
+      )}
+      <button onClick={nextGreeting}>🎲</button>
+      {children}
+    </>
+  );
+}
+
+```
+
+```js Copyright.js
+export default function Copyright({year}) {
+  return <p>©️ {year}</p>;
+}
+```
+
+```js greetings.js
+export default [
+    {type: 'text', value: 'Howdy!'},
+    {type: 'text', value: 'Hello!'},
+    {type: 'image', value: 'https://i.imgur.com/yXOvdOSs.jpg'},
+    {type: 'image', value: 'https://i.imgur.com/yXOvdOSs.jpg'},
+    {type: 'text', value: 'Howdy!'},
+];
+```
+
+```css
+.fancy {
+    font-style: italic;
+    color: blue;
+}
+```
+</Sandpack>
+
+<Diagram name="conditional_render_tree" height={300} width={522} alt="TODO.">
+
+With conditional rendering, across different renders, the render tree may render different components.
+
+</Diagram>
+
+In this example, depending on what `greeting.type` is, we may render `<Title>` or `<Image>`. The render tree may be different for each render pass.
+
+Although render trees may differ across render pases, these trees are generally helpful for identifying what the top-level components are in a React app. Top-level components affect the rendering performance of all the components beneath them and often contain the most complexity.
+
+## The Module Dependency Tree {/*the-module-dependency-tree*/}
```

Comment:
I think it might be helpful to say somewhere in this section that importing a lot of code that you don't use can sometimes impact how fast your app runs.

Reply:
expanded on this in the last paragraph in this section